apm-agent-plugins/apm-api-plugin/src/test/java/co/elastic/apm/agent/pluginapi/TransactionInstrumentationTest.java

```diff
@@ -193,7 +193,7 @@ public void startSpan() {
     public void testAgentPaused() {
         // end current transaction first
         endTransaction();
-        reporter.reset();
+        reporter.resetWithoutRecycling();
```

Comment:
[suggestion/question] I think it could be relevant to add another boolean in `MockReporter` like  `checkDestinationAddress` or `checkUnknownOutcomes` to optionally disable the recycling check and always call `reset()`, because here it's not that we don't want to trigger recycling, it's just that we need to not test that everything is recycled.

Reply:
It's the other way around - `reset()` does the assertion and then calls `resetWithoutRecycling()`, I just removed the assertion.
Why do I need it to be recycled and not dropped?

Reply:
discussed it live, my idea was to only have a single `reset` method that always recycle, but only assert for proper recycling like we do for outcome of span type check : assertion enabled by default, but we can opt-out on a per-test basis. We will leave that for later refactor.